webapp/src/ts/services/recurring-process-manager.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+
+import { RelativeDateService } from './relative-date.service';
+import { UnreadRecordsService } from './unread-records.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecurringProcessManagerService {
+
+  private recurringProcesses:any = {
+    updateRelativeDates: false,
+    updateReadDocsCount: false
+  };
+
+  constructor(
+    private relativeDateService: RelativeDateService,
+    private unreadRecordsService: UnreadRecordsService
+  ) { }
+
+  private stopRecurringProcess(processName) {
+    if (this.recurringProcesses[processName]) {
+      clearInterval(this.recurringProcesses[processName]);
+      this.recurringProcesses[processName] = false;
+    }
+  }
+
+  startUpdateRelativeDate() {
+    if (this.recurringProcesses.updateRelativeDates) {
+      return;
+    }
+
+    this.recurringProcesses.updateRelativeDates = setInterval(
+      this.relativeDateService.updateRelativeDates,
```

Comment:
This is throwing an error in the console. 
Change to `() => this.relativeDateService.updateRelativeDates()` to resolve context correctly.

Reply:
yep, fixed